thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineTaskRunner.java

```diff
@@ -138,8 +137,12 @@ public DetectionPipelineTaskRunner() {
         this.evaluationDAO.save(evaluationDTO);
       }
 
-      // run maintenance flow to update model
-      config = maintenanceFlow.maintain(config, Instant.now());
+      try {
+        // run maintenance flow to update model
+        config = maintenanceFlow.maintain(config, Instant.now());
+      } catch (Exception e) {
+        LOG.warn("Re-tune pipeline {} failed", config.getId(), e);
```

Comment:
Let's add a counter here.

Reply:
Maybe better to add a counter in all calls to tune().

Reply:
Added the counter. The counter will increment when the tuner is called